tests/system-tests/rdscore/internal/rdscorecommon/metallb-traffic-segregation.go

```diff
@@ -0,0 +1,744 @@
+package rdscorecommon
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/netip"
+	"strings"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/golang/glog"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-gotests/tests/system-tests/internal/remote"
+	"github.com/openshift-kni/eco-gotests/tests/system-tests/internal/supporttools"
+
+	. "github.com/openshift-kni/eco-gotests/tests/system-tests/rdscore/internal/rdscoreinittools"
+	"github.com/openshift-kni/eco-gotests/tests/system-tests/rdscore/internal/rdscoreparams"
+)
+
+const (
+	lbDeploymentLabel     = "systemtest-test=rds-core-metal-app"
+	stNamespace           = "rds-metallb-supporttools-ns"
+	stDeploymentLabel     = "rds-core=supporttools-deploy"
+	packetCaptureProtocol = "http"
+	codesPattern          = "200 404"
+	captureScript         = `#!/bin/bash
+tcpdump -vvv -e -nn -i %s | egrep -i 'Host:'
+`
+)
+
+var (
+	bgpOneAppURLIPv4 = fmt.Sprintf("http://%s:%s",
+		RDSCoreConfig.MetalLBLoadBalancerOneIPv4, RDSCoreConfig.MetalLBTrafficSegregationTargetPort)
+	bgpOneAppURLIPv6 = fmt.Sprintf("http://[%s]:%s",
+		RDSCoreConfig.MetalLBLoadBalancerOneIPv6, RDSCoreConfig.MetalLBTrafficSegregationTargetPort)
+	bgpTwoAppURLIPv4 = fmt.Sprintf("http://%s:%s",
+		RDSCoreConfig.MetalLBLoadBalancerTwoIPv4, RDSCoreConfig.MetalLBTrafficSegregationTargetPort)
+	bgpTwoAppURLIPv6 = fmt.Sprintf("http://[%s]:%s",
+		RDSCoreConfig.MetalLBLoadBalancerTwoIPv6, RDSCoreConfig.MetalLBTrafficSegregationTargetPort)
+)
+
+// VerifyMetallbEgressTrafficSegregation test metallb egress traffic segregation.
+//
+//nolint:funlen
+func VerifyMetallbEgressTrafficSegregation(ctx SpecContext) {
+	By("Asserting if test URLs are provided")
+
+	if RDSCoreConfig.MetalLBTrafficSegregationTargetOneIPv4 == "" &&
+		RDSCoreConfig.MetalLBTrafficSegregationTargetOneIPv6 == "" {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"Test URLs for MetalLB FRR testing not specified or are empty. Skipping...")
+		Skip("Test URL for MetalLB FRR testing not specified or are empty")
+	}
+
+	if RDSCoreConfig.MetalLBTrafficSegregationTargetTwoIPv4 == "" &&
+		RDSCoreConfig.MetalLBTrafficSegregationTargetTwoIPv6 == "" {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"Test URLs for the second MetalLB FRR testing not specified or are empty. Skipping...")
+		Skip("Test URL for the second MetalLB FRR testing not specified or are empty")
+	}
+
+	By("Randomly choosing node for the support-tools deployment")
+
+	segregationTestNode, err := getNodeForTest()
```

Comment:
this one assumes that any worker node is configured to learn BGP routes which is not always the case. Therefor I suggest to adjust function to accept a parameter that will allow to select particular node(s), for e.g.:
```suggestion
	segregationTestNode, err := getNodeForTest(RDSCoreConfig.MetalLB2ndEgressNodeLabel)
```

Reply:
the idea was different when this method was implemented. let me revert it to the original version

Reply:
small refactor was done, please take a look on the latest code version